python/pyspark/sql/tests/connect/test_parity_dataframe.py

```diff
@@ -57,13 +57,11 @@ def test_create_nan_decimal_dataframe(self):
     def test_duplicated_column_names(self):
         super().test_duplicated_column_names()
 
-    # TODO(SPARK-41871): DataFrame hint parameter can be a float
```

Comment:
https://github.com/apache/spark/blob/56f8dccd6532aaebf9f2838be6673d5c75c6df79/python/pyspark/sql/tests/test_dataframe.py#L536-L561

```
        plan = df1.join(df2.hint("broadcast"), "id")._jdf.queryExecution().executedPlan()
        self.assertEqual(1, plan.toString().count("BroadcastHashJoin"))
```


```
        logical_plan = hinted_df._jdf.queryExecution().logical()

        self.assertEqual(1, logical_plan.toString().count("1.2345"))
        self.assertEqual(1, logical_plan.toString().count("what"))
        self.assertEqual(3, logical_plan.toString().count("itworks"))
```


the two tests use `._jdf.queryExecution()` to get the string repr of plan to check the hints, but is not able to do similar check in Connect.